eeg_flow/utils/concurrency.py

```diff
@@ -20,7 +20,7 @@ def lock_files(*args, timeout: float = 10) -> List[SoftFileLock]:
         The list of locks. When deleted, a lock is automatically released.
     """
     check_type(timeout, ("numeric",), "timeout")
-    files = [ensure_path(arg, must_exist=False) for arg in args]
+    files = [ensure_path(arg, must_exist=True) for arg in args]
```

Comment:
Why did you change this?
I think it should stay to `False`, IIRC, this is used to lock the derivative files at the beginning of a processing step, which by definition are not yet created.

Reply:
for the very first step of converting the xdf to fif, I was thinking that we should first put a lock on the xdf? 
anyway, probably best that I return to a pseudocode/flowchart, to make sure I have the right idea in mind?

Reply:
The lock is required on files you write too, not on files you read; so I don't think it's needed.
If person 1 starts to work on xdf A, it will lock the output files derived from xdf A.
If person 2 starts to work simultaneously to work on xdf A, it will detect the lock of the *outputs* and prevent him from working on the same file.

Reply:
ah I see thanks! will test that now